avro/src/main/java/com/fasterxml/jackson/dataformat/avro/deser/AvroParserImpl.java

```diff
@@ -320,7 +320,7 @@ protected final void convertNumberToInt() throws IOException
     protected final void convertNumberToLong() throws IOException
     {
         if ((_numTypesValid & NR_INT) != 0) {
-            _numberLong = (long) _numberInt;
```

Comment:
I sort of like explicitness of casts like this, even if not minimal. But I guess that's ok to remove, technically speaking.